core/src/main/java/net/opentsdb/query/processor/merge/MergerFactory.java

```diff
@@ -171,15 +200,31 @@ public TypedTimeSeriesIterator newIterator(final Merger node,
                                                final QueryResult result,
                                                final Collection<TimeSeries> sources,
                                                final TypeToken<? extends TimeSeriesDataType> type) {
-      return new MergerNumericArrayIterator(node, result, sources);
+      switch (((MergerConfig) node.config()).getMode()) {
+        case HA:
+          return new MergerNumericArrayIterator(node, result, sources);
+        case SPLIT:
+          return new SplitNumericArrayIterator(node, result, sources);
+        default:
+          throw new IllegalArgumentException("No implementation for mode: "
+                  + ((MergerConfig) node.config()).getMode());
+      }
     }
 
     @Override
     public TypedTimeSeriesIterator newIterator(final Merger node,
                                                final QueryResult result,
                                                final Map<String, TimeSeries> sources,
                                                final TypeToken<? extends TimeSeriesDataType> type) {
-      return new MergerNumericArrayIterator(node, result, sources);
+      switch (((MergerConfig) node.config()).getMode()) {
+        case HA:
+          return new MergerNumericArrayIterator(node, result, sources);
+        case SPLIT:
```

Comment:
If we dont implement the iterator for split, is the merge already done and the result represented as a single HA ?

Reply:
No, it would throw an NPE I believe. The Numeric and Summary will for now.